regression_tests/basisComparator.cpp

```diff
@@ -0,0 +1,128 @@
+
```

Comment:
I suppose the copyright heading should be added to the cpp files (multiple new files in this PR).